cacheops/simple.py

```diff
@@ -46,8 +48,20 @@ def wrapper(*args, **kwargs):
                 cache_key = 'c:' + _get_key(func, args, kwargs, extra)
                 try:
                     result = self.get(cache_key)
+
                 except CacheMiss:
-                    result = func(*args, **kwargs)
+                    if USE_LOCK:
+                        # This will block in all calls but the
+                        # first to avoid dog-pile effect.
+                        with redis_lock.Lock(self.conn, cache_key):
```

Comment:
As I use the `cached()` decorator in [1flow](/1flow/1flow) and really need the dog-pile-avoiding feature, I propose the test be `if USE_LOCK and getattr(self, 'conn', None):`. What do you think about it ?